alf/networks/disjoint_policy_value_network.py

```diff
@@ -0,0 +1,263 @@
+                alf.nest.flatten(projection_net))
+            self._actor_branch = alf.nn.Sequential(
+                self._actor_encoder,
+                alf.nn.Branch(projection_net, name='NestedProjection'))
+        else:
+            self._actor_branch = alf.nn.Sequential(self._actor_encoder,
+                                                   projection_net)
+
+        # +------------------------------------------+
+        # | Step 3: Value head of the aux branch     |
+        # +------------------------------------------+
+
+        # Note that the aux branch value head belongs to the policy component.
+
+        # Like the value head Aux head is outputing value estimation
+        self._aux_branch = alf.nn.Sequential(
+            self._actor_encoder,
+            alf.layers.FC(
+                input_size=self._actor_encoder.output_spec.shape[0],
+                output_size=1), alf.layers.Reshape(shape=()))
+
+        # +------------------------------------------+
+        # | Step 4: Value head of the value branch   |
+        # +------------------------------------------+
+
+        if is_sharing_encoder:
+            # Use the same encoder, but the encoder is DETACHED.
+            self._value_branch = alf.nn.Sequential(
+                self._actor_encoder, alf.layers.Detach(),
+                alf.layers.FC(
+                    input_size=self._actor_encoder.output_spec.shape[0],
+                    output_size=1), alf.layers.Reshape(shape=()))
+        else:
+            # If not sharing encoder, create a separate encoder for the value
+            # component.
+            value_encoder = encoding_network_ctor(
+                input_tensor_spec=observation_spec,
+                kernel_initializer=kernel_initializer)
+
+            self._value_branch = alf.nn.Sequential(
+                value_encoder,
+                alf.layers.FC(
+                    input_size=self._actor_encoder.output_spec.shape[0],
+                    output_size=1), alf.layers.Reshape(shape=()))
+
+    def forward(self, observation, state: DisjointPolicyValueNetworkState):
+        """Computes the action distribution, aux value and value estimation
+
+        Args:
+
+            observation (nested torch.Tensor): a tensor that is consistent with
+                the encoding network
+
+            state: the state for RNN based network for interface compatibility,
+                which is always set to empty in this case because this network
+                does not have an RNN-based encoder
+
+        Returns:
+            act_dist (torch.distributions): action distribution
+            state: empty
+
+        """
+
+        action_distribution, actor_state = self._actor_branch(
+            observation, state=state.actor)
+
+        value, value_state = self._value_branch(observation, state=state.value)
+
+        aux, aux_state = self._value_branch(observation, state=state.aux)
```

Comment:
when sharing encoder, actor_encoder is called twice with the same input. It's better to call it once.

Reply:
Good catch! Sorry I missed this. Will update.